SparkPost/2web2ui
# SA-2078: Validate URL query params in report builder

### What Changed

- Cleaned up query params before being saved into context
- Added a validator for filters.

### How To Test

- Go to `signals/analytics`.
- Change one of the metrics to an invalid one. It should be removed and the user alerted.
- Change the timezone to an invalid value. It should default to local timezone and the user alerted.
- Change the date to an invalid value. Might also need to change range to `custom` It should default to 7 days ad alert the user.
- Change the filter to an invalid value. It should remove an invalid filter at the top level and alert the user. This means that if you have 3 AND filters but only the first one is invalid, the ui will keep the other 2.
